sdk/cloudmachine/Azure.Projects.OpenAI/src/OpenAIModelFeature.cs

```diff
@@ -145,7 +120,12 @@ protected override ProvisionableResource EmitResources(ProjectInfrastructure cm)
             deployment.DependsOn.Add(previousDeployment);
         }
 
-        cm.AddResource(deployment);
+        cm.AddConstruct(deployment);
+
+        string key = Kind == AIModelKind.Chat ? "OpenAI.Chat.ChatClient" : "OpenAI.Embeddings.EmbeddingClient";
+        string locator = Kind == AIModelKind.Chat ? $"{cm.ProjectId}_chat" : $"{cm.ProjectId}_embedding";
```

Comment:
should we make these switch statements so that if another Kind gets added we'll have a default case where we can throw?

Reply:
There will be so many changes all over the place when we add a new model type. I am not sure the throw adds enough value.